### DISCUSSION TITLE
How to update a session in V5?

### DISCUSSION THREAD

**Discussion by maxpaleo** [nextauthjs/next-auth#12419] [Help]
How on earth do we update a session in V5? 

The new docs don't mention anything related to updating sessions, the only reference is the [UpdateSession()](https://authjs.dev/reference/nextjs/react#updatesession) from auth/react. 

auth/react exports the `useSession` hook, which contains the `update` function, which in previous versions caused the `update` trigger to be available in the jwt callbacks, however, nothing is sent in V5. 

Is the update function in useSession still accurate? 

Tbh, its unbelievable how frustrating it is to work with auth.js. I mean, every single thing requires hours of research only to find out the issue is either related to bugs or inaccurate documentation. 

I get it, V5 is in beta, but isn't updating the session one of the core elements of authentication?

**Comment by ghost** [comment]
I can't guarantee this solution will work, but I found this in my search as well. I haven't had a chance to test it yet, since I moved to clerk. Try this, and let me know if it works :) 

[nextauth-v5 session update](https://medium.com/@youngjun625/next-js14-nextauth-v5-2-session-update-b977cb6afd47)

**Comment by PAAPII10** [comment]
# Using `unstable_update` in NextAuth Beta

You can use the `unstable_update` helper from NextAuth to update the session server-side.

## 1. Import `unstable_update`

```ts
export const { handlers, signIn, signOut, auth, unstable_update } = NextAuth({
  // your config
});
```
## 2. Create a server function to update the session
Inside a `"use server"` file, define a helper like this:
```ts
"use server";

import { auth, unstable_update } from "@/auth"; // adjust path accordingly
import type { Session } from "next-auth";

export const updateSession = async ({
  data,
}: {
  data: Partial<Session["user"]>;
}) => {
  const session = await auth();
  if (session) {
    await unstable_update({
      ...session,
      user: {
        ...session.user,
        ...data,
      },
    });
  }
};
```

## 3. Call updateSession wherever you need to refresh/update session data
For example, after refreshing access tokens:

```ts
const newToken = await arvasitAuth.refreshAccessToken({
  token: refreshToken,
});

if (newToken.accessToken) {
  setCookie("accessToken", newToken.accessToken);
  setCookie("refreshToken", newToken.refreshToken);

  await updateSession({
    data: {
      accessToken: newToken.accessToken,
      refreshToken: newToken.refreshToken,
    },
  });
}
```

**Comment by OahMada** [comment]
The v4 doc still applies
https://next-auth.js.org/getting-started/client#updating-the-session
